BattingAvgCalc.py

```diff
@@ -0,0 +1,32 @@
+from pyspark import keyword_only
+from pyspark.ml import Transformer
+from pyspark.ml.param.shared import HasInputCols, HasOutputCol
+from pyspark.ml.util import DefaultParamsReadable, DefaultParamsWritable
+
+
+class BattingAvgCalc(
+    Transformer,
+    HasInputCols,
+    HasOutputCol,
+    DefaultParamsReadable,
+    DefaultParamsWritable,
+):
+    @keyword_only
+    def __init__(self, inputCols=None, outputCol=None):
+        super(BattingAvgCalc, self).__init__()
+        kwargs = self._input_kwargs
+        self.setParams(**kwargs)
+        return
+
+    @keyword_only
+    def setParams(self, inputCols=None, outputCol=None):
+        kwargs = self._input_kwargs
+        return self._set(**kwargs)
+
+    def _transform(self, dataset):
+        input_cols = self.getInputCols()
+        output_col = self.getOutputCol()
+        dataset = dataset.withColumn(
+            output_col, dataset[input_cols[0]] / dataset[input_cols[1]]
+        )
+        return dataset
```

Comment:
I am not sure if this is the way Julien asked us to implement the Transformer. We have to calculate the batting average using the Transformer, that's what the assignment says.

Reply:
Julien has mentioned to me 
Just wrap one of your steps (the one that calculates the BA) and wrap that into a transformer step to prove you know how to do it.
So i thought that it is ok to just use any one part of calculation in transform. 
I am trying to get it confirmed from Julien

Reply:
Yeah, this looks good.